ph5/utilities/set_deploy_pickup_times.py

```diff
@@ -85,91 +83,199 @@ def get_args():
-                of.write("\tdeploy_time/micro_seconds_i = %d\n" %
-                         dep_time.micro_seconds_i)
+                if dep_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tdeploy_time/micro_seconds_i=%d\n" %
+                             dep_time[inc].micro_seconds_i)
             elif post == 'type_s':
-                of.write("\tdeploy_time/type_s = %s\n" % dep_time.type_s)
+                if dep_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tdeploy_time/type_s=%s\n" %
+                             dep_time[inc].type_s)
             elif post == 'ascii_s':
-                of.write("\tdeploy_time/ascii_s = %s\n" % dep_time.ascii_s)
+                if dep_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tdeploy_time/ascii_s=%s\n" %
+                             dep_time[inc].ascii_s)
         elif pickupRE.match(line):
             key, val = line.split('=')
             pre, post = key.split('/')
             post = post.strip()
             if post == 'epoch_l':
-                of.write("\tpickup_time/epoch_l = %d\n" % pu_time.epoch_l)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/epoch_l=%d\n" %
+                             pu_time[inc].epoch_l)
             elif post == 'micro_seconds_i':
-                of.write("\tpickup_time/micro_seconds_i = %d\n" %
-                         pu_time.micro_seconds_i)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/micro_seconds_i=%d\n" %
+                             pu_time[inc].micro_seconds_i)
             elif post == 'type_s':
-                of.write("\tpickup_time/type_s = %s\n" % pu_time.type_s)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/type_s=%s\n" % pu_time[inc].type_s)
+                if auto is True:
+                    inc = inc+1  # Increments the list of input units.
+                else:
+                    inc = 0
             elif post == 'ascii_s':
-                of.write("\tpickup_time/ascii_s = %s\n" % pu_time.ascii_s)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/ascii_s=%s\n" %
+                             pu_time[inc].ascii_s)
         else:
             of.write("\t%s\n" % line)
 
 
+def kefTables(kef_file, NICKNAME, PATH):
+
+    k = kefx.Kef(ARRAY_FILE)
+    k.open()
+    while True:
+        n = k.read(10000)
+        if n == 0:
+            LOGGER.error("Kef file is empty")
```

Comment:
Should this really be logged as an error?  I think `n` will eventually be 0 when the file is finished being read.

Reply:
Perhaps there should be a check to see if the file is empty before it is processed.

Reply:
Commits 896cee4